admin_interface/models.py

```diff
@@ -401,10 +415,26 @@ class Theme(models.Model):
 
     objects = ThemeQuerySet.as_manager()
 
+    @property
+    def static_logo_relative_url(self):
+        return os.path.relpath(self.static_logo_path, settings.STATIC_ROOT)
+
     def set_active(self):
         self.active = True
         self.save()
 
+    def save(self, *args, **kwargs):
+        if self.static_logo_path:
+            # clear cache if static logo path has changed
+            try:
+                obj = Theme.objects.get(pk=self.pk)
```

Comment:
This query (and the related logic) can be avoided by checking if the `logo_static` path value is different from it's initial value: https://stackoverflow.com/a/1793323/2096218